WcaOnRails/app/views/competitions/show.html.erb

```diff
@@ -1,5 +1,23 @@
 <% provide(:title, @competition.name) %>
 
-<%= render layout: 'results_nav' do %>
-  <%= render "results_table", results: @competition.winning_results, hide_pos: true, hide_round: true %>
+<%= render layout: 'nav' do %>
+  <ul class="nav nav-tabs">
+    <li><a href="#general-info" data-toggle="tab"><%=t '.show.general_info' %></a></li>
+    <% @competition.competition_tabs.each do |tab| %>
+      <li><a href="#<%= "#{tab.id}-#{tab.name.parameterize}" %>" data-toggle="tab"><%= tab.name %></a></li>
+    <% end %>
+  </ul>
+
+  <div class="tab-content" id="competition-tabs">
+    <div class="tab-pane" id="general-info">
+      <%= render layout: 'results_nav', locals: { layout_nav: false } do %>
+        <%= render "results_table", results: @competition.winning_results, hide_pos: true, hide_round: true %>
+      <% end %>
+    </div>
+    <% @competition.competition_tabs.each do |tab| %>
+      <div class="tab-pane" id="<%= "#{tab.id}-#{tab.name.parameterize}" %>">
```

Comment:
This `id` must be unique but should be readable (it appears in the url). This was my idea at first, but I'd be glad to see if there's something nicer.